flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -340,6 +362,65 @@ public void resetNFAChanged() {
 		return Tuple2.of(result, timeoutResult);
 	}
 
+	private void discardComputationStatesAccordingToStrategy(Queue<ComputationState<T>> computationStates,
```

Comment:
The changes in the next commit are not enough. The `Map` type should be also changed in `SharedBuffer#extractPatterns`. Also please provide tests for this behaviour (returning results in Pattern order), so that it will not be possible to change it by mistake.